python/ray/data/_internal/execution/operators/actor_pool_submitter.py

```diff
@@ -0,0 +1,181 @@
+from typing import Dict, Any, Iterator, Callable, List, Tuple
+import ray
+from ray.data.block import Block, BlockAccessor, BlockMetadata, BlockExecStats
+from ray.data.context import DEFAULT_SCHEDULING_STRATEGY, DatasetContext
+from ray.data._internal.compute import ActorPoolStrategy
+from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
+from ray.data._internal.execution.operators.map_task_submitter import MapTaskSubmitter
+from ray.types import ObjectRef
+
+
+class ActorPoolSubmitter(MapTaskSubmitter):
+    """A task submitter for MapOperator that uses a Ray actor pool."""
+
+    def __init__(
+        self, compute_strategy: ActorPoolStrategy, ray_remote_args: Dict[str, Any]
```

Comment:
Shall we remove the compute_strategy dependency and pass in the individual parameters separately?

Reply:
We need to determine an appropriate pool size to use, since `compute_strategy.max_size` isn't always the best choice (e.g. it will be `float("inf")` by default), but I can pull that logic out into the caller.